### PR TITLE
Fix Flag Finding

### PR DESCRIPTION
**Pull request by Spedwards**

### CODE REVIEW

File: creep.action.dropping.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ action.newTarget = function(creep) {
         drop = creep.pos.findClosestByRange(creep.room.structures.spawns);
     }
     if( !drop ) {
-        drop = creep.pos.findClosestByRange(creep.room.find(FIND_FLAGS, {filter: FLAG_COLOR.claim.spawn.filter}));
+        drop = creep.pos.findClosestByRange(creep.room.find(FIND_FLAGS, Flag.flagFilter(FLAG_COLOR.commands.drop)));
```
Comment (chobobobo) [inline creep.action.dropping.js]: Yeah, this needs to filter for FLAG_COLOR.claim.spawn, I will remove the change from the other PR.

**Comment by karlthepagan** [comment]
@chobobobo is this ok now?

File: room.js [hunk 2]
```diff
@@ -404,8 +404,9 @@ mod.extend = function(){
                 get: function() {
                     if( _.isUndefined(this._piles) ){
                         const room = this.room;
-                        this._piles = room.find(FIND_FLAGS, {filter: FLAG_COLOR.command.drop.filter})
+                        this._piles = FlagDir.filter(FLAG_COLOR.command.drop, this.room.getPositionAt(25,25), true)
                             .map(function(flag) {
+                                flag = Game.flags[flag.name];
```
Comment (chobobobo) [inline room.js]: I don't think this line is necessary as it's a parameter to the function.

File: room.js [hunk 3]
```diff
@@ -404,8 +404,9 @@ mod.extend = function(){
                 get: function() {
                     if( _.isUndefined(this._piles) ){
                         const room = this.room;
-                        this._piles = room.find(FIND_FLAGS, {filter: FLAG_COLOR.command.drop.filter})
+                        this._piles = FlagDir.filter(FLAG_COLOR.command.drop, this.room.getPositionAt(25,25), true)
```
Comment (chobobobo) [inline room.js]: nitpicking at this point, but `this.room` could be changed to `room` as it's a const above.

File: room.js (hunk 2 above)
Comment (chobobobo) [inline room.js]: Ah, nevermind, I remember you pointed out that FlagDir doesn't return flags, it returns some other data object.. maybe rename the parameter to something other than flag so other people don't get confused like I do when skimming the code

Comment (Spedwards) [inline room.js]: As I mentioned in Slack, `FlagDir.filter` does not return flags, rather an object containing flag information.

**Pull request merged**